Ajout projet et modification en hook pour les call apollo

Taloud opened the pull request.

Taloud committed:
Ajout de la notion de projet et modification en hook pour les call apollo

Taloud committed:
modification unique key for multiple project

kwiss commented on app/components/Articles/EditArticle.js:
```diff
@@ -0,0 +1,73 @@
+import { FaTrashAlt,FaCheck, FaBan } from 'react-icons/fa'
+
+
+const DELETE_ARTICLE = gql`
```
les const de query/mutation sont a bouger dans un fichier correspondant

kwiss commented on app/components/Articles/EditArticle.js:
```diff
@@ -0,0 +1,73 @@
+        >
+          <button className={props.enabled ? "deleteButton" : "deleteButton createButton" } type="submit"> {props.enabled ? <FaBan /> : <FaCheck/> }
+      </button>
+        </form>
```
strange indent do you use eslint ?

kwiss commented on app/components/Articles/PaginationArticle.js:
```diff
@@ -1,37 +1,37 @@
+    <div className="pagination">
            <Link href={{
-              pathname: '/',
+              pathname: '/project/'+ projectId,
```
?? `/project/${projectId}`

revolunet replied:
template literals FTW !

kwiss commented on app/components/PageLayout.js:
```diff
@@ -11,6 +11,8 @@ export default props => (
       <meta name="viewport" content="initial-scale=1.0, width=device-width" />
       <link rel="stylesheet" type="text/css" href="/static/nprog.css" />
+      <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous"/>
+
```
😔

kwiss commented on app/components/Projects/AddProject.js:
```diff
@@ -0,0 +1,56 @@
+export default function AddProject() {
+  let inputName;
+  let inputDescription;
+  const [addProject, { data }] = useMutation(ADD_PROJECT);
```
data ?

kwiss commented on app/components/Research/ElasticResearch.js:
```diff
@@ -0,0 +1,178 @@
+                            "size":5,"_source":{"includes":["*"],"excludes":[]},"from":0}									
+                            
+                    }
+                    }
```
move that function

kwiss commented on app/components/Research/ElasticResearch.js:
```diff
@@ -0,0 +1,178 @@
+                                    }
+                               
+                            }
+                        }
```
move that in another forlder

kwiss commented:
globally, there is a problem with indent you should use eslint https://github.com/SocialGouv/linters

kwiss reviewed (changes requested):
check indent, console.log etc for merge

Taloud committed:
fix indent + console.log + queries files and some functions

kwiss commented on app/.eslintrc.json:
```diff
@@ -0,0 +1,23 @@
+        "es6": true,
+        "node": true
+    },
+    "extends": "eslint:recommended",
```
you should use https://github.com/SocialGouv/linters

kwiss commented on app/components/Research/ElasticResearch.js:
```diff
@@ -1,178 +1,130 @@
+                ]
+            }
+        },
+        "size": 5, "_source": { "includes": ["*"], "excludes": [] }, "from": 0
```
indent is really strange

Commit:
feat: add docker-compose setup

 - add /elastic and /graphql-engine proxies into Next.js server. this reduce complexity of dealing with various hostnames
 - add sample docker-compose.override.prod.yaml to enable app build in production
 - add app/Dockerfile

by running `docker-compose up` one should be able to get the full setup

Commit:
feat: add socialgouv linters

revolunet commented:
Hugo, je t'invite à relire la doc sur les commits conventionnels : https://www.conventionalcommits.org

voir aussi la PR sur les linters

Taloud committed:
Merge pull request #3 from SocialGouv/linters

feat: add linter

Taloud committed:
Merge pull request #4 from SocialGouv/docker-compose-setup

feat: docker-compose

Taloud committed:
feat: add third status to article
BREAKING CHANGE: resolve all conflict on apollo hook and lint

Commit:
fix: add config for server in apollo file

Taloud committed:
fix: missing ';' in config

Commit:
fix: add networking for local hasura on docker

Commit:
fix: remove docker network

Commit:
fix urls

Commit:
fixes

Taloud committed:
Merge pull request #5 from SocialGouv/deployment

fix: docker-compose

Commit:
doc: add README

Commit:
feat: add yarn lint

Taloud committed:
Merge pull request #6 from SocialGouv/readme

doc: add README

Taloud committed:
Merge pull request #7 from SocialGouv/lint

feat: add yarn lint

Taloud committed:
docs: test push

Taloud committed:
FEAT: add filter to research and track change for texte

Taloud committed:
FIX: good const for elastic
